src/main/java/duke/logic/parsers/ParserUtil.java

```diff
@@ -0,0 +1,149 @@
+package duke.logic.parsers;
+
+import duke.commons.MessagesPrompt;
+import duke.commons.exceptions.DukeDateTimeParseException;
+import duke.commons.exceptions.DukeException;
+import duke.commons.Messages;
+import duke.commons.exceptions.DukeUnknownCommandException;
+import duke.model.events.Deadline;
+import duke.model.events.DoWithin;
+import duke.model.events.Event;
+import duke.model.events.Todo;
+
+import java.time.LocalDateTime;
+
+/**
+ * Parser for utility functions.
+ */
+public class ParserUtil {
+    /**
+     * Parses the userInput and return a new to-do constructed from it.
+     *
+     * @param userInput The userInput read by the user interface.
+     * @return The new to-do object.
+     */
+    protected static Todo createTodo(String userInput) throws DukeException {
+        String description = userInput.substring("todo".length()).strip();
+        if (description.isEmpty()) {
+            throw new DukeUnknownCommandException();
+        }
+        return new Todo(description);
+    }
+
+    /**
+     * Parses the userInput and return a new deadline constructed from it.
+     *
+     * @param userInput The userInput read by the user interface.
+     * @return The new deadline object.
+     */
+    protected static Deadline createDeadline(String userInput) throws DukeException {
```

Comment:
don't you think this is doing more than utility functionality

Reply:
Code deleted